dir/fs.go

```diff
@@ -0,0 +1,159 @@
+package dir
+
+import (
+	"errors"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+// rootedFS is io.FS implementation used in New.
+// root is the root of the file system tree passed to os.DirFS.
+type rootedFS struct {
+	fs.FS
+	Root string
+}
+
+// NewRootedFS create a rootedFS
+func NewRootedFS(fsys fs.FS, root string) rootedFS {
+	return rootedFS{fsys, root}
+}
```

Comment:
It seems this `NewRootedFS` method is not used anywhere. You may consider removing it.

Reply:
I use this method in plugin/manager/manager_test.go to new a memory based vfs.